src/main/java/de/muenchen/isi/domain/service/calculation/InfrastrukturbedarfService.java

```diff
@@ -421,269 +421,229 @@ protected Stream<PersonenProJahrModel> calculatePersonenForWohneinheitProJahr(
         final var personen = new ArrayList<PersonenProJahrModel>();
         final var numberOfYearsToCalculate = 20 - (Integer.parseInt(wohneinheiten.getJahr()) - earliestYear);
         var yearToCalculate = 0;
-        BigDecimal anzahlPersonenGesamt;
+        PersonenProJahrModel anzahlPersonen;
         if (yearToCalculate < numberOfYearsToCalculate) {
-            anzahlPersonenGesamt =
-                wohneinheiten
-                    .getWohneinheiten()
-                    .multiply(
-                        sobonOrientierungswertSozialeInfrastruktur.getObererRichtwertEinwohnerJahr1NachErsterstellung()
-                    )
-                    .setScale(CalculationService.DIVISION_SCALE, RoundingMode.HALF_UP);
-            personen.add(
-                createPersonenProJahr(Integer.parseInt(wohneinheiten.getJahr()) + yearToCalculate, anzahlPersonenGesamt)
-            );
+            anzahlPersonen =
+                calculatePersonenWithObererRichtwertAndWohneinheitenForCertainYear(
+                    wohneinheiten,
+                    sobonOrientierungswertSozialeInfrastruktur.getObererRichtwertEinwohnerJahr1NachErsterstellung(),
+                    yearToCalculate
+                );
+            personen.add(anzahlPersonen);
             yearToCalculate++;
```

Comment:
Der obere Richtwert ist ja abhängig vom Jahr. Wäre es nicht besser, die Kalkulation in einer Loop zu machen und abhängig vom yearToCalculate den oberen Richtwert aus sobonOrientierungswertSozialeInfrastruktur zu holen?

Reply:
Die konkatenierten If-Statements sind durch eine Schleife ersetzt.